sql/sql_table.cc

```diff
@@ -9588,9 +9588,13 @@ enum fk_column_change_type
 
 static enum fk_column_change_type
 fk_check_column_changes(THD *thd, Alter_info *alter_info,
-                        List<LEX_CSTRING> &fk_columns,
-                        const char **bad_column_name)
+                        FOREIGN_KEY_INFO *fk,
+                        const char **bad_column_name,
+                        bool referenced=false)
 {
+  List<LEX_CSTRING> fk_columns= referenced
+                                ? fk->referenced_fields
+                                : fk->foreign_fields;
```

Comment:
I believe that the ternary operator should be indented by exactly 2 additional spaces (in total, 4 columns from the start).